### ISSUE TITLE
The psnr_avg of `ficus` is abnormal, others are similar to the reported results

### ISSUE THREAD

**Issue by MaxChanger** [KAIR-BAIR/nerfacc#76]
Hi Ruilong,

Thanks for the amazaing project. I tested the three methods you provided, and I can get similar results on the corresponding datasets, but only one scene gets lower performance, and the loss is nan. The results are consistent across multiple runs.
Do you have any tips on how to find the cause of the problem.

RTX 3090, torch=1.10.1+cu111, torchvision=0.11.2+cu111
nerfacc == 0.2.0

```shell
python examples/train_mlp_nerf.py --train_split train --scene ficus
elapsed_time=1.78s | step=0 | loss=0.00888 | alive_ray_mask=64 | n_rendering_samples=10231 | num_rays=64 |
elapsed_time=163.39s | step=5000 | loss=nan | alive_ray_mask=0 | n_rendering_samples=65611 | num_rays=269 |
elapsed_time=325.57s | step=10000 | loss=nan | alive_ray_mask=0 | n_rendering_samples=60619 | num_rays=260 |
elapsed_time=487.55s | step=15000 | loss=nan | alive_ray_mask=0 | n_rendering_samples=63759 | num_rays=274 |
elapsed_time=649.66s | step=20000 | loss=nan | alive_ray_mask=0 | n_rendering_samples=65520 | num_rays=281 |
elapsed_time=812.03s | step=25000 | loss=nan | alive_ray_mask=0 | n_rendering_samples=66233 | num_rays=273 |
elapsed_time=975.32s | step=30000 | loss=nan | alive_ray_mask=0 | n_rendering_samples=69855 | num_rays=279 |
elapsed_time=1138.03s | step=35000 | loss=nan | alive_ray_mask=0 | n_rendering_samples=65633 | num_rays=276 |
elapsed_time=1300.72s | step=40000 | loss=nan | alive_ray_mask=0 | n_rendering_samples=66092 | num_rays=273 |
elapsed_time=1462.58s | step=45000 | loss=nan | alive_ray_mask=0 | n_rendering_samples=64994 | num_rays=278 |
elapsed_time=1624.28s | step=50000 | loss=nan | alive_ray_mask=0 | n_rendering_samples=72772 | num_rays=288 |
100%|█████████████████████████████████| 200/200 [1:31:58<00:00, 27.59s/it]
evaluation: psnr_avg=14.225740423202515
training stops
````

![image](https://user-images.githubusercontent.com/35401968/196374918-a793dc37-c7d9-49f8-a635-61c0d02b6464.png)

**Comment by liruilong940607** [comment]
Thanks for reporting! I will investigate it soon. My hunch is that might be an unfortunate initialization as that scene have a lot of empty areas

**Comment by liruilong940607** [comment]
Unfortunately I can't reproduce the issue you report. Things behave normally on my end.

<img width="1329" alt="Screen Shot 2022-10-19 at 5 30 35 PM" src="https://user-images.githubusercontent.com/10151885/196829252-f1129041-ebce-427f-9914-697a7d6aa0cb.png">

**Comment by liruilong940607** [comment]
I noticed for the first iteration, the `n_rendering_samples` are slightly different between your run and mine. This is weird as the random seed have been fixed here:

https://github.com/KAIR-BAIR/nerfacc/blob/f9be5ad3b181058e27d484ac2e441973e797db18/examples/train_mlp_nerf.py#L23

So I would expect you to see exactly the same log with mine for at least the first iteration as the network should be initialization with exactly the same weights. Not sure what is happening here. Maybe try with the latest code in this repo?

**Comment by MaxChanger** [comment]
Thanks for your advice and help. Changing only the seed without changing any code really makes a difference. Now it looks like psnr_avg makes sense.
first image seed=40, second image seed=1024.   
seed=42 will produce `nan` on my machine, just like the log above

<img width="1488" alt="image" src="https://user-images.githubusercontent.com/35401968/196850234-34e73841-e33a-41ca-bd87-36aeda43715a.png">

<img width="1488" alt="image" src="https://user-images.githubusercontent.com/35401968/196850192-7df0dd19-8244-4026-acd0-8b92a62f5ff1.png">

**Comment by liruilong940607** [comment]
Yeah the reason is by default the first iteration only samples 64 rays (for memory management). If it is unfortunate that all those rays end up at empty regions then the network would learn only nonsense. Ficus is a scene that has lots of empty regions so makes it easier to happen.

Solution to this problem can be either:
1/ Have more rays in the start of the training
2/ use a warmup schedule for learning rate.

We didn't do that as we don't want to make the example scripts too complicated. But it is definitely recommend to have a warmup learning rate schedule for your nerf training.

**Comment by MaxChanger** [comment]
Thanks for your time, andI'll dive into the details of the code implementation.
